src/components/item-display-panel/item-display-panel.component.ts

```diff
@@ -1,19 +1,19 @@
-import { Component, Directive, Input, SimpleChange, Output, EventEmitter } from '@angular/core';
+import { Component, Directive, Input, SimpleChange, Output, EventEmitter, Renderer2, ElementRef, Host } from '@angular/core';
 
 @Component({
     selector: 'ux-item-display-panel',
     templateUrl: './item-display-panel.component.html',
     host: {
         '(document:click)': 'clickOff($event)',
-        '(document:keyup.escape)': 'visible = false'
+        '(document:keyup.escape)': 'visible = false',
+        '[class.inline-host]' : 'inline', 
+        '[class.visible-host]' : 'visible' 
```

Comment:
You currently are running code in ngOnChanges to update the top position. This should probably be moved to a host property instead